### PR TITLE
label event thread, to make debugging easier

### PR DESCRIPTION
**Pull request by thegecko**
## Changes

-

## Fixes

-

## Checklist

Have you...

- [ ] Tested the change acts as expected
- [ ] Checked the change doesn't remove or change existing functionality
- [ ] Considered how the feature impacts the product and tested around it (not just the happy paths)
- [ ] Where possible, executed the hardware tests on multiple operating systems

### CODE REVIEW

File: src/thread_name.cc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    // For Windows, we use the new SetThreadDescription API which
+    // is only available in newish versions. To avoid taking any
+    // risks (and because on certain versions it's the only
+    // option to access the API), we use the dynamic linker
+    #include <windows.h>
+
+    extern "C" typedef HRESULT (WINAPI *SetThreadDescriptionFn)(HANDLE, PCWSTR);
+
+    static auto RetrieveSymbol(const char* objectName) {
```
Comment (mildsunrise) [inline src/thread_name.cc:46]:
```suggestion
    static SetThreadDescriptionFn RetrieveSymbol(const char* objectName) {
```

**Comment by mildsunrise** [comment]
tested on Windows and it works correctly as well. it only needed one change, which I have submitted as a suggestion

**Comment by mildsunrise** [comment]
I think we can merge this by now that it's been tested in the 3 OSes (the function does nothing in other OSes)

**Comment by thegecko** [comment]
I've not tested this at all. Is there an easy way to test it on mac/win?

**Comment by mildsunrise** [comment]
yes it's simple to test yourself, just attach a debugger to a Node.js process after `require('usb')`, and you should see a thread named `node-usb events`. if you see it, it's working correctly :)

I don't think we can test it though, but it's fine since it's only a debugging aid

**Pull request merged**